x-pack/platform/plugins/shared/alerting/public/pages/maintenance_windows/components/recurring_schedule_field.test.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { PropsWithChildren } from 'react';
+import { screen, render, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Frequency } from '@kbn/rrule';
+import { Form, useForm, useFormData } from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+import type { RecurringSchedule } from '@kbn/response-ops-recurring-schedule-form/types';
+import { RecurringScheduleField } from './recurring_schedule_field';
+import type { FormProps } from './schema';
+import { schema } from './schema';
+import { RecurrenceEnd } from '@kbn/response-ops-recurring-schedule-form/constants';
+
+const baseValue = {
+  title: 'Test',
+  startDate: '2025-01-01',
+  endDate: '2025-01-02',
+  recurring: true,
+};
+
+const TestWrapper = ({
+  defaultValue = {
+    ...baseValue,
+    recurringSchedule: {
+      frequency: Frequency.YEARLY,
+      ends: RecurrenceEnd.ON_DATE,
+    },
+  },
+  onFormDataChange,
+  children,
+}: PropsWithChildren<{
+  defaultValue?: FormProps;
+  onFormDataChange?: (data: FormProps) => void;
+}>) => {
+  const { form } = useForm<FormProps>({
+    defaultValue,
+    options: { stripEmptyFields: false },
+    schema,
+  });
+
+  useFormData<FormProps>({
+    form,
+    onChange: onFormDataChange,
+  });
+
+  return <Form form={form}>{children}</Form>;
+};
+
+describe('RecurringScheduleField', () => {
```

Comment:
I think we should make sure here that conflicts between `TestWrapper` form configuration and `RecurringScheduleForm` configuration are handled as expected.

What if the outer form has a default value, and `RecurringScheduleForm` gets passed default values as props.
Or, how are conflicts between outer and inner schema handled?

(This is connected to my comment on `recurring_schedule_field.tsx`)